pkg/entities/tokens.go

```diff
@@ -91,3 +79,32 @@ func (e *Entity) GetAllSupportedToken(guildID string) (returnToken []model.Token
 
 	return returnToken, nil
 }
+
+func getCoinGeckoBySymbol(symbol string) (*response.CoinGekoResponse, error) {
+	req, err := http.NewRequest(http.MethodGet, "https://api.coingecko.com/api/v3/coins/list", nil)
```

Comment:
try using `SearchCoin` in `coingecko.go` instead

Reply:
resolved